src/android/com/microsoftopentechnologies/intellij/wizards/createvm/SelectImageStep.java

```diff
@@ -0,0 +1,312 @@
+
+        model.configStepList(createVmStepsList, 1);
+
+        final ArrayList imageTypeList = new ArrayList();
+        imageTypeList.add("Public Images");
+        imageTypeList.addAll(Arrays.asList(PublicImages.values()));
+        imageTypeList.add("MSDN Images");
+        imageTypeList.addAll(Arrays.asList(MSDNImages.values()));
+        imageTypeList.add("Private Images");
+        imageTypeList.addAll(Arrays.asList(PrivateImages.values()));
+
+
+        imageTypeComboBox.setModel(new DefaultComboBoxModel<Object>(imageTypeList.toArray()) {
+            @Override
+            public void setSelectedItem(Object o) {
+                if (o instanceof Enum) {
+                    super.setSelectedItem(o);
+                }
+            }
+        });
+
+        imageTypeComboBox.setRenderer(new DefaultListCellRenderer() {
+            @Override
+            public Component getListCellRendererComponent(JList<?> jList, Object o, int i, boolean b, boolean b1) {
+                if (o instanceof Enum) {
+                    return super.getListCellRendererComponent(jList, "  " + o.toString(), i, b, b1);
+                } else {
+                    JLabel label = new JLabel(o.toString());
+                    Font f = label.getFont();
+                    label.setFont(f.deriveFont(f.getStyle()
+                            | Font.BOLD | Font.ITALIC));
+                    return label;
+                }
+            }
+        });
+
+        imageTypeComboBox.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent itemEvent) {
+                fillList();
+            }
+        });
+
+        imageLabelList.setCellRenderer(new DefaultListCellRenderer() {
+            @Override
+            public Component getListCellRendererComponent(JList<?> jList, Object o, int i, boolean b, boolean b1) {
+                String cellValue = (o instanceof VirtualMachineImage) ? ((VirtualMachineImage) o).getLabel() : o.toString();
+                this.setToolTipText(cellValue);
+                return super.getListCellRendererComponent(jList, cellValue, i, b, b1);
+            }
+        });
+
+        imageLabelList.addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                VirtualMachineImage virtualMachineImage = (VirtualMachineImage) imageLabelList.getSelectedValue();
+                model.setVirtualMachineImage(virtualMachineImage);
+
+                if(virtualMachineImage != null) {
+                    imageDescriptionTextPane.setText(model.getHtmlFromVMImage(virtualMachineImage));
+                    imageDescriptionTextPane.setCaretPosition(0);
+                    model.getCurrentNavigationState().NEXT.setEnabled(true);
+                }
+            }
+        });
+
+        imageDescriptionTextPane.addHyperlinkListener(new HyperlinkListener() {
```

Comment:
This snippet of code for handling hyperlinks in the VM image description seems to be repeating in multiple steps. Does it make sense to refactor this into a common `VMWizardBaseStep` class?